src/api/app/controllers/cloud/upload_jobs_controller.rb

```diff
@@ -48,5 +54,11 @@ def upload_data
         to_h.
         merge(user: ::User.current)
     end
+
+    def set_upload_job
+      @upload_job = ::Cloud::User::UploadJob.find_by(job_id: params[:id])
```

Comment:
```
user.upload_jobs.find(params[:id])
```
?

Reply:
Hm, does not reflect if the user is admin :/

Reply:
That's checked below. We can not verfify that earlier because the owner of the upload job should get access as well.